stdlib/UUIDs/src/UUIDs.jl

```diff
@@ -81,4 +83,34 @@ function uuid4(rng::AbstractRNG=Random.GLOBAL_RNG)
     UUID(u)
 end
 
+"""
+    uuid5(ns::UUID, name::String) -> UUID
+
+Generates a version 5 (namespace and domain-based) universally unique identifier (UUID),
+as specified by RFC 4122.
+
+# Examples
+```jldoctest
+julia> rng = MersenneTwister(1234);
+
+julia> u4 = uuid4(rng)
+UUID("196f2941-2d58-45ba-9f13-43a2532b2fa8")
+
+julia> u5 = uuid5(u4, "julia")
+UUID("6f461186-52d8-5fc1-993a-77a729165b65")
+```
+"""
+function uuid5(ns::UUID, name::String)
+    hash_result = SHA.sha1(string(ns,name))
+    # set version number to 5
+    hash_result[7] = (hash_result[7] & 0x0F) | (0x50)
+    hash_result[9] = (hash_result[9] & 0x3F) | (0x80)
+    v = zero(UInt128)
+    #use only the first 16 bytes of the SHA1 hash
+    for idx in Base.OneTo(16)
```

Comment:
Why not just `1:16`?

Reply:
OneTo is generally, in some applications lots of UUIDs can get generated.
I'll build a benchmark if the argument is needed